lib/api/sibling-nav.js

```diff
@@ -4,22 +4,23 @@ import { queryAPI } from "@/lib/fetch";
 export async function addSiblings(entryData, site = "default") {
   if (!entryData) return null;
 
-  const { uri, level } = entryData;
-
-  const siblings = await getSiblings(uri, level, site);
-
+  const { uri, level, parent } = entryData;
+  const parentId = parent?.id;
+  const siblings = parentId
+    ? await getSiblings(parentId, uri, level, site)
+    : {};
   return Object.assign(entryData, { siblings });
 }
 
-export async function getSiblings(uri, level = 1, site = "default") {
+export async function getSiblings(parentId, uri, level = 1, site = "default") {
   const query = gql`
       {
         siblings: entry (uri: "${uri}", site: "${site}") {
-            prev(section: "pages", site: "${site}", level: ${level}) {
+            prev(descendantOf: ${parentId}, section: "pages", site: "${site}", level: ${level}) {
                 uri
                 title
             }
-            next(section: "pages", site: "${site}", level: ${level}) {
+            next(descendantOf: ${parentId}, section: "pages", site: "${site}", level: ${level}) {
```

Comment:
In a perfect world the underlying Craft shenanigans ought to be constraining this `prev` and `next` already, but appears they are not.  The edge case seems to be weird stuff like entries who share a parent but it's not a "direct" parent, or entries used to be siblings but were reshuffled in the pagetree, and some other stuff like that.  Adding this filter seems to affirm strongly enough to only return "direct" siblings 🤷